app/routes/items/staff_conversation.js

```diff
@@ -0,0 +1,38 @@
+import AuthorizeRoute from "./../authorize";
+import Ember from "ember";
+
+export default AuthorizeRoute.extend({
+  i18n: Ember.inject.service(),
+  canManageItemMessages: Ember.computed.alias(
+    "session.currentUser.canManageItemMessages"
+  ),
+
+  model(params) {
+    return Ember.RSVP.hash({
+      item: this.loadIfAbsent("item", params.item_id),
+      messages: this.get("canManageItemMessages")
+        ? this.store.query("message", {
+            package_id: params.item_id,
+            is_private: true
+          })
+        : []
+    });
+  },
+
+  setupController(controller, model) {
+    this._super(controller, model);
+    controller.set("messages", model.messages);
+    controller.set("model", model.item);
+
+    if (this.get("canManageItemMessages")) {
```

Comment:
`on` and `off` methods are meant to be lifecycle hooks. So they should always be called. But the conditional `this.get("canManageItemMessages")` can exist within those methods to do X or Y